dask/order.py

```diff
@@ -115,786 +106,427 @@ def order(
+                                        branch = path.copy()
+                                        branch.append(d)
+                                        branches.append(branch)
+                                break
+                            linear_hull.update(deps_downstream)
+                            path.extend(sorted(deps_downstream, key=sort_key))
+                            continue
+                        elif current in known_runnable_paths:
+                            known_runnable_paths[current].append(path)
+                            if (
+                                len(known_runnable_paths[current])
+                                >= num_needed[current]
+                            ):
+                                pruned_branches: deque[list[Key]] = deque()
+                                for path in known_runnable_paths.pop(current):
+                                    if path[-2] not in result:
+                                        pruned_branches.append(path)
+                                if len(pruned_branches) < num_needed[current]:
+                                    known_runnable_paths[current] = list(
+                                        pruned_branches
+                                    )
+                                else:
+                                    if validate:
+                                        nodes_in_branches = set()
+                                        for b in pruned_branches:
+                                            nodes_in_branches.update(b)
+                                        cond = not (
+                                            dependencies[current]  # type: ignore
+                                            - set(result)
+                                            - nodes_in_branches
+                                        )
+                                        assert cond
+                                    while pruned_branches:
+                                        path = pruned_branches.popleft()
+                                        for k in path:
+                                            if num_needed[k]:
+                                                pruned_branches.append(path)
+                                                break
+                                            add_to_result(k)
+                        else:
+                            if (
+                                len(dependencies[current]) > 1  # type: ignore
+                                and num_needed[current] <= 1
+                            ):
+                                for k in path:
+                                    add_to_result(k)
+                            else:
+                                known_runnable_paths[current] = [path]
+                        break
 
-    result: dict[Key, int] = {}
-    i = 0
+    def pick_strategy() -> bool:
```

Comment:
Nit: I'd suggest naming the strategies and returning their name here. A boolean result of `pick_strategy` is meaningless without additional context.

Reply:
I renamed it to `use_longest_path` to make it explicit. I kind of like the bool version better for now considering that we don't have many strategies and am not totally convinced that I want many